pyomo/contrib/pyros/util.py

```diff
@@ -1731,6 +1761,77 @@ def process_termination_condition_master_problem(config, results):
             )
 
 
+def call_solver(model, solver, config, timing_obj, timer_name, err_msg):
+    """
+    Solve a model with a given optimizer, keeping track of
+    wall time requirements.
+
+    Parameters
+    ----------
+    model : ConcreteModel
+        Model of interest.
+    solver : Pyomo solver type
+        Subordinate optimizer.
+    config : ConfigDict
+        PyROS solver settings.
+    timing_obj : TimingData
+        PyROS solver timing data object.
+    timer_name : str
+        Name of sub timer under the hierarchical timer contained in
+        ``timing_obj`` to start/stop for keeping track of solve
+        time requirements.
+    err_msg : str
+        Message to log through ``config.progress_logger.exception()``
+        in event an ApplicationError is raised while attempting to
+        solve the model.
+
+    Returns
+    -------
+    SolverResults
+        Solve results. Note that ``results.solver`` contains
+        an additional attribute, named after
+        ``TIC_TOC_SOLVE_TIME_ATTR``, of which the value is set to the
+        recorded solver wall time.
+
+    Raises
+    ------
+    ApplicationError
+        If ApplicationError is raised by the solver.
+        In this case, `err_msg` is logged through
+        ``config.progress_logger.exception()`` before
+        the excception is raised.
+    """
+    tt_timer = TicTocTimer()
+
+    orig_setting, custom_setting_present = adjust_solver_time_settings(
+        timing_obj, solver, config
+    )
+    timing_obj.start_timer(timer_name)
+    tt_timer.tic(msg=None)
+
+    try:
+        results = solver.solve(
+            model, tee=config.tee, load_solutions=False, symbolic_solver_labels=True
```

Comment:
Historically there were performance implications with setting `symbolic_solver_labels=True`. I'm not sure if these performance implications are still relevant with many of the performance improvements John has implemented over the last few years. Have you compared the runtime of PyROS with this option on and off to check if this has a noticeable impact?

If it is significant you might want to make this configurable instead of a hard-coded option. You could also have a "debug" mode for PyROS that automatically turns it on.

Reply:
Understood. I have looked into comparing runtimes for a collection of 12 instances (each with ~5000 variables and constraints, 4--5 PyROS iterations required per instance, ~60 subordinate solver calls per instance), solved with IPOPT and GAMS/CONOPT (through the `IPOPT` and `GAMSShell` interfaces) as subordinate solvers. For these instances, running PyROS with `symbolic_solver_labels=True` instead of `symbolic_solver_labels=False` appears to incur an additional wall time requirement of ~30s (3--5% of total runtime) on average. The additional wall time requirement incurred per instance may be more pronounced for instances requiring more PyROS iterations, as the solve time requirements for the larger subproblems, which grow in size as the iteration number is increased, are most affected.

I am considering adding an optional argument `symbolic_solver_labels=False` to `PyROS.solve()` to make the subordinate solver calls more configurable.

Reply:
@shermanjasonaf, what would you like to do here:
- merge as is?
- switch the hard-coded `symbolic_solver_labels` to `False`?
- add in the new (advanced?) configuration option?

Reply:
I have added a new configuration option; see subsequent commits.